qiskit/compiler/assembler.py

```diff
@@ -343,13 +352,21 @@ def _parse_common_args(
         for lo_config in schedule_los
     ]
 
+    measure_esp_enabled = getattr(backend_config, "measure_esp_enabled", False)
+    if use_measure_esp is None:
+        use_measure_esp = True if measure_esp_enabled else False  # default use of esp readout
+    if not measure_esp_enabled and use_measure_esp:
+        raise QiskitError("ESP readout not supported on this device. Please make sure the flag "
+                          "'use_measure_esp' is set to 'False' or not used.")
```

Comment:
```suggestion
                          "'use_measure_esp' is unset or set to 'False'.")
```

Reply:
Addressed in https://github.com/Qiskit/qiskit-terra/pull/6543/commits/de6d6073315d5607ef59c929861b38aa6d47847a.